### PR TITLE
[12.0][MIG] shared_governance_base

### PR DESCRIPTION
**Pull request by carmenbianca**
Task: https://gestion.coopiteasy.be/web#id=7883&view_type=form&model=project.task&action=475&active_id=210

### CODE REVIEW

File: shared_governance_base/models/mandate.py [hunk 1]
```diff
@@ -1,11 +1,10 @@
-# -*- coding: utf-8 -*-
 from odoo import api, fields, models
 
 
 class Mandate(models.Model):
     _name = "sg.mandate"
 
-    _inherit = ["mail.thread", "ir.needaction_mixin"]
+    _inherit = ["mail.thread", "mail.activity.mixin"]
```
Comment (carmenbianca) [inline shared_governance_base/models/mandate.py]: Does this _just work_, or do I need to add some migratory stuff somewhere?

Comment (huguesdk) [reply shared_governance_base/models/mandate.py]: from what i understand by reading the comment [here](https://github.com/OCA/OCB/blob/10.0/odoo/addons/base/ir/ir_needaction.py), only inheriting from `ir.needaction_mixin` without defining the corresponding methods has no effect, so we might as well just remove it. what are the reasons to add `mail.activity.mixin`?

File: shared_governance_base/__manifest__.py [hunk 2]
```diff
@@ -1,11 +1,10 @@
-# -*- coding: utf-8 -*-
 #
 #  Copyright (C) 2019  Coop IT Easy SCRLfs. <http://www.coopiteasy.be>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 {
     "name": "Shared Governance",
-    "version": "10.0.1.0.0",
+    "version": "11.0.1.0.0",
     "depends": ["base", "mail", "document"],
```
Comment (huguesdk) [inline shared_governance_base/__manifest__.py]: `base` is always implied, i think, and `document` is not used here.

File: shared_governance_base/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#  Copyright (C) 2019  Coop IT Easy SCRLfs. <http://www.coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Shared Governance",
+    "version": "12.0.1.0.0",
+    "depends": ["base", "mail", "document"],
+    "author": "Coop IT Easy SCRLfs",
+    "category": "Cooperative Management",
+    "website": "www.coopiteasy.be",
```
Comment (huguesdk) [inline shared_governance_base/__manifest__.py]: it’s not a correct url. running pre-commit should fix this.

File: shared_governance_base/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        Shared Governance in Sociocracy organizations.
+    """,
+    "data": [
+        "security/shared_governance_security.xml",
+        "security/ir.model.access.csv",
+        "views/mandate.xml",
+        "views/res_partner.xml",
+        "views/menus.xml",
+    ],
+    "installable": True,
```
Comment (huguesdk) [inline shared_governance_base/__manifest__.py]: `installable` is true by default, thus should be removed.

**Review by huguesdk — CHANGES_REQUESTED** [review]
please base this new `12.0` branch on https://github.com/coopiteasy/cie-repository-template to have the correct configuration files (for pre-commit etc.). sorry for not telling you this earlier. before doing this, if you could update the configuration in that repo to match the changes you did in other repos, that would be great.

**Comment by carmenbianca** [comment]
Deploy note to self: Add vertical sociocracy to gitaggregate and odoo.conf

File: shared_governance_base/__manifest__.py (hunk 3 above)
Comment (carmenbianca) [reply shared_governance_base/__manifest__.py]: no pre-commit in this repo ;) will fix

File: shared_governance_base/models/mandate.py (hunk 1 above)
Comment (carmenbianca) [reply shared_governance_base/models/mandate.py]: > what are the reasons to add mail.activity.mixin?

From what little research I did, `mail.activity.mixin` is the designated successor of `ir.needaction_mixin`. But if it's superfluous all the same, I can remove it.

Comment (huguesdk) [reply shared_governance_base/models/mandate.py]: i think we need input from an analyst on this then.

**Comment by carmenbianca** [comment]
> before doing this, if you could update the configuration in that repo to match the changes you did in other repos, that would be great.

@huguesdk I will not do this now, but I will get around to doing this. It's on my todo list for the implementation of CI.

File: shared_governance_base/models/mandate.py (hunk 1 above)
Comment (robinkeunen) [reply shared_governance_base/models/mandate.py]: The history of module: we did a hackaton during a _mise au vert_ retreat. We probably had in mind to add some actions to be followed in the mandates but did not go all the way to implement them.
=> remove it

**Pull request closed without merging**